File: blog/domain/icon.py
"""Domain models for the Icon entity."""

from typing import override
from dataclasses import dataclass


@dataclass
class Icon:
    """Domain model for blog icon."""

    id: int | None = None
    title: str = ""
    url: str = ""
    content: str | None = None

    @override
    def __str__(self):
        return f"Icon(id={self.id}, title={self.title})"

File: rust/crates/persistence/src/entities/icon.rs
//! `icons` table entity.

use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "icons")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    #[sea_orm(column_type = "String(sea_orm::sea_query::StringLen::N(255))", unique)]
    pub title: String,
    #[sea_orm(column_type = "String(sea_orm::sea_query::StringLen::N(255))", unique)]
    pub url: String,
    #[sea_orm(column_type = "Text", nullable)]
    pub content: Option<String>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
